infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerPhase.java

```diff
@@ -45,4 +51,32 @@ public BrokerPhase then(BrokerPhase next) {
    */
   @Beta
   public abstract List<ChePlugin> execute() throws InfrastructureException;
+
+  /**
+   * Start a new span using provided tracer. If tracer is null, returns null.
+   *
+   * <p>Meant to be used with {@link BrokerPhase#finishSpanIfExists(Span)}
+   */
+  public Span startTracingPhase() {
+    if (tracer == null) {
```

Comment:
Maybe it is better to use your new tracing util?

Reply:
It's indeed *a lot* cleaner to do that -- don't know why I didn't start there. These methods on `BrokerPhase` are totally unnecessary.